### PR TITLE
Makes create_slots automatically setup weights for swap_weights

### PR DESCRIPTION
**Pull request by marksandler2**
# Description

Brief Description of the PR:

Fixes # (issue)  https://github.com/tensorflow/addons/issues/2107

## Type of change

- [x ] Bug fix 

# Checklist:

- [x ] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running pre-commit hooks
- [x ] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?

Run unit-tests

### CODE REVIEW

File: tensorflow_addons/optimizers/moving_average.py [hunk 1]
```diff
@@ -127,11 +127,14 @@ def get_config(self):
         return {**base_config, **config}
 
     def _create_slots(self, var_list):
-        self._optimizer._create_slots(
-            var_list=var_list
-        )  # pylint: disable=protected-access
+        self._optimizer._create_slots(var_list=var_list)  # pylint: disable=protected-access
```
Comment (WindQAQ) [inline tensorflow_addons/optimizers/moving_average.py:130]: we don't use pylint anymore, can you remove the comment `# pylint: xxx`? Thanks!

File: tensorflow_addons/optimizers/tests/moving_average_test.py [hunk 2]
```diff
@@ -228,6 +228,39 @@ def test_dynamic_decay():
     np.testing.assert_allclose(ema_var0.read_value(), [0.64, 1.64])
 
 
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+@pytest.mark.with_device([tf.distribute.MirroredStrategy])
+def test_swap_weight_no_shadow_copy():
```
Comment (WindQAQ) [inline tensorflow_addons/optimizers/tests/moving_average_test.py:233]:
```suggestion
def test_swap_weight_no_shadow_copy(device):
```

**Review by WindQAQ — COMMENTED** [review]
Thank you! Seems that it fails our coding style. Can you run `bash tools/pre-commit.sh` to format the codes? Thank you again for the fix!

File: tensorflow_addons/optimizers/moving_average.py (hunk 1 above)
Comment (marksandler2) [reply tensorflow_addons/optimizers/moving_average.py]: Gotcha. Removed

File: tensorflow_addons/optimizers/moving_average.py [hunk 3]
```diff
@@ -29,6 +29,7 @@ class MovingAverage(AveragedOptimizerWrapper):
     Empirically it has been found that using the moving average of the trained
     parameters of a deep network is better than using its trained parameters
     directly. This optimizer allows you to compute this moving average and swap
+    raise app.UsageError('Too many command-line arguments.')
```
Comment (WindQAQ) [inline tensorflow_addons/optimizers/moving_average.py:32]: This is added accidentally?

**Review by WindQAQ — APPROVED** [review]
Thank you!

**Pull request merged**